src/test/tpc/tpcc_test.cpp

```diff
@@ -0,0 +1,461 @@
+      insert->set_transaction_context(transaction_context);
+      insert->execute();
+      transaction_context->commit();
+    }
+  }
+
+  std::shared_ptr<const Table> get_validated(const std::string& table_name,
+                                             const std::shared_ptr<TransactionContext>& transaction_context) {
+    auto get_table = std::make_shared<GetTable>(table_name);
+    get_table->execute();
+
+    auto validate = std::make_shared<Validate>(get_table);
+    validate->set_transaction_context(transaction_context);
+    validate->execute();
+
+    return validate->get_output();
+  }
+
+  void verify_table_sizes(const std::unordered_map<std::string, size_t>& sizes) {
+    for (const auto& [table_name, size] : sizes) {
+      auto transaction_context = Hyrise::get().transaction_manager.new_transaction_context();
+      const auto table = get_validated(table_name, transaction_context);
+
+      if (table_name == "ORDER_LINE" && size == 0) {
+        // For ORDER_LINE, the number of lines per order is in [5, 15] (see 4.3.3.1) and non-deterministic. If it is
+        // not specifically calculated, we just check if it is within the acceptable range.
+        EXPECT_GE(table->row_count(), sizes.at("ORDER") * 5) << "Failed table: " << table_name;
+        EXPECT_LE(table->row_count(), sizes.at("ORDER") * 15) << "Failed table: " << table_name;
+      } else {
+        EXPECT_EQ(table->row_count(), size) << "Failed table: " << table_name;
+      }
+    }
+  }
+
+  // See TPC-C Spec 1.2.1
+  const std::unordered_map<std::string, size_t> initial_sizes{{"WAREHOUSE", NUM_WAREHOUSES},
+                                                              {"DISTRICT", NUM_WAREHOUSES * 10},
+                                                              {"HISTORY", NUM_WAREHOUSES * 30'000},
+                                                              {"STOCK", NUM_WAREHOUSES * 100'000},
+                                                              {"NEW_ORDER", NUM_WAREHOUSES * 9'000},
+                                                              {"CUSTOMER", NUM_WAREHOUSES * 30'000},
+                                                              {"ITEM", 100'000},
+                                                              {"ORDER_LINE", 0},  // see verify_table_sizes
+                                                              {"ORDER", NUM_WAREHOUSES * 30'000}};
+
+  static std::unordered_map<std::string, BenchmarkTableInfo> tables;
+  static constexpr auto NUM_WAREHOUSES = 2;
+};
+
+std::unordered_map<std::string, BenchmarkTableInfo> TPCCTest::tables;
+
+TEST_F(TPCCTest, InitialTables) { verify_table_sizes(initial_sizes); }
+
+TEST_F(TPCCTest, Delivery) {
+  // As the procedures have some internal logic that we do not want to replicate in the tests (e.g., picking a W_ID),
+  // we first create a transaction that has a view on the unmodified database, then execute the procedure, and finally
```

Comment:
```suggestion
  // we first create a transaction context that has a view on the unmodified database, then execute the procedure, and finally
```

Reply:
I wouldn't say so. It is the transaction that has the view, not the context.